packages/shared/src/components/notifications/NotificationItem.tsx

```diff
@@ -0,0 +1,84 @@
+import classNames from 'classnames';
+import createDOMPurify from 'dompurify';
+import React, { ReactElement, ReactNode, useEffect, useRef } from 'react';
+import { getFadedBackground } from '../../lib/styling';
+import NotificationItemAttachment, {
+  NotificationItemAttachmentProps,
+} from './NotificationItemAttachment';
+import NotificationItemAvatar, {
+  NotificationItemAvatarProps,
+} from './NotificationItemAvatar';
+
+interface NotificationItemProps {
+  icon: ReactNode;
+  title: string;
+  isUnread?: boolean;
+  description?: string;
+  avatars?: NotificationItemAvatarProps[];
+  attachments?: NotificationItemAttachmentProps[];
+}
+
+function NotificationItem({
+  icon,
+  title,
+  isUnread,
+  description,
+  avatars,
+  attachments,
+}: NotificationItemProps): ReactElement {
+  const DOMPurify = useRef<DOMPurify.DOMPurifyI>();
+
+  useEffect(() => {
+    DOMPurify.current = createDOMPurify(window);
+  }, []);
```

Comment:
We already use this somewhere correct?
Perhaps we can convert it into a hook (just for clean code)

Reply:
If it's reused a hook might still be best, but it's not a hard requirement, so leaving it up to you.

Reply:
Cool. Was just checking if you are cool too with creating a hook to abstract this single useState. In my opinion, we should go with your suggestion so I will do that before moving forward. Thank you!